lib/matplotlib/axes/_base.py

```diff
@@ -3902,18 +3902,18 @@ def _make_twin_axes(self, *kl, **kwargs):
 
     def twinx(self):
         """
-        Create a twin Axes sharing the xaxis
+        Create a twin Axes instance sharing the x-axis with self
 
-        Create a twin of Axes for generating a plot with a shared
-        x-axis but independent y-axis.  The y-axis of self will have
-        ticks on left and the returned axes will have ticks on the
-        right. To ensure tick marks of both axis align, see
-        `~matplotlib.ticker.LinearLocator`
+        Create a twin Axes instance for generating a plot with a shared
```

Comment:
If the docstring is going to focus on implementation, it still needs to be revised to reflect what's actually going on behind the scenes. twinx/twiny doesn't really "share" an axis in the sense of sharing the `Axis` object, but instead creates a 2nd, invisible one..I can try to encapsulate this aspect and revise.